### PR TITLE
[G7T5-42] Edit BE API for skill

### PR DESCRIPTION
**Pull request by branfats**
# Description
Edit backend API for skill to call DB

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Not Applicable

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backend/app/api/api_v1/endpoints/skill.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    Create new skill.
+    """
+    if not settings.USERS_OPEN_REGISTRATION:
+        raise HTTPException(
+            status_code=403,
+            detail="Open staff registration is forbidden on this server",
+        )
+    skill_in = schemas.SkillCreate(
+        skill_name=skill_name,
+    )
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/skill.py:53]: I don't think these tally with the Skills' attributes?

File: backend/app/api/api_v1/endpoints/skill.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    skill = crud.skill.get(db, skill_id=skill_id)
+    return skill
+
+
+@router.post("", response_model=schemas.Skill)
+def create_skill(
+    *,
+    db: Session = Depends(deps.get_db),
+    skill_name: str = Body(...),
+) -> Any:
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/skill.py:42]: Should take in the other necessary attributes to create a skill

File: backend/app/models/skill.py [hunk 3]
```diff
@@ -8,4 +8,4 @@ class Skill(Base):
     course_id = Column(String, ForeignKey("course.course_id"), nullable=False) 
     skill_name = Column(String, nullable=False)
     skill_desc = Column(String)
-    is_Active = Column(Boolean)
+    is_active = Column(Boolean)
```
Comment (jonathantan1425) [inline backend/app/models/skill.py:11]: Good catch

**Review by jonathantan1425 — CHANGES_REQUESTED** [review]
The attributes should align with the models that you're referencing

**Comment by jonathantan1425** [comment]
Can you add the unit tests as well? Following this:
- [Endpoint test](https://github.com/is212g7t5/spm/blob/master/backend/app/tests/api/api_v1/test_role.py)
- [Crud test (to database)](https://github.com/is212g7t5/spm/blob/master/backend/app/tests/crud/test_role.py)
- [Utils if necessary](https://github.com/is212g7t5/spm/blob/master/backend/app/tests/utils/role.py)

File: backend/app/api/api_v1/endpoints/skill.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    skill = crud.skill.create(db, obj_in=skill_in)
+    return skill
+
+
+@router.put("/{skill_id}", response_model=schemas.Skill)
+def update_skill_by_id(
+    *,
+    db: Session = Depends(deps.get_db),
+    skill_id: int,
+    skill_name: str = Body(None),
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/skill.py:69]: Missing the remaining skill attributes

File: backend/app/tests/crud/test_staff.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    skill_update = SkillUpdate(
+        skill_name=new_skill_name,
+        skill_desc=skill.skill_desc,
+        is_active=skill.is_active,
+    )
+    skill2 = crud.skill.update(db=db, db_obj=skill, obj_in=skill_update)
+    assert skill2.skill_id == skill.skill_id
+    assert skill2.skill_desc == skill.skill_desc
+    assert skill2.skill_name == new_skill_name
+
```
Comment (jonathantan1425) [inline backend/app/tests/crud/test_staff.py:40]: /nit Need to assert is_active?

File: backend/app/api/api_v1/endpoints/skill.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    skill_desc: str = Body(None),
+    is_active: bool,
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/skill.py:71]: if `Body(None)` default is not provided, will this work?

**Review by jonathantan1425 — COMMENTED** [review]
Small question on the is_active default for create and update - will the endpoint work if the value is provided and not provided?

File: backend/app/tests/crud/test_staff.py (hunk 5 above)
Comment (branfats) [reply backend/app/tests/crud/test_staff.py:40]: Sorry this was a duplicated file. Will delete this in the next commit.
The file should be test_skill not test_staff

backend/app/tests/crud/test_skill.py

Comment (branfats) [reply backend/app/tests/crud/test_staff.py:40]: Sorry this was a duplicated file. Will delete this in the next commit.
The file should be test_skill not test_staff

backend/app/tests/crud/test_skill.py

Comment (branfats) [reply backend/app/tests/crud/test_staff.py:40]: Sorry this was a duplicated file. Will delete this in the next commit.
The file should be test_skill not test_staff

backend/app/tests/crud/test_skill.py

Comment (branfats) [reply backend/app/tests/crud/test_staff.py:40]: But regarding assert is_active, I'm not sure if need can someone advise?

Comment (branfats) [reply backend/app/tests/crud/test_staff.py:40]: Don't think so, backend/app/tests/crud/test_job.py didn't have assert is_active as well

File: backend/app/api/api_v1/endpoints/skill.py (hunk 6 above)
Comment (branfats) [reply backend/app/api/api_v1/endpoints/skill.py:71]: I think it wouldn't work. I've included it in next commit

File: backend/app/api/api_v1/endpoints/skill.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+
+@router.post("", response_model=schemas.Skill)
+def create_skill(
+    *,
+    db: Session = Depends(deps.get_db),
+    skill_id: int,
+    skill_name: str = Body(...),
+    skill_desc: str = Body(...),
+    is_active: bool,
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/skill.py:44]: Should have the Body(...) default here as well

**Review by jonathantan1425 — APPROVED** [review]

**Pull request merged**